go/interpreter/lfvm/interpreter_intergration_test.go

```diff
@@ -0,0 +1,689 @@
+func validFrom(revision tosca.Revision) revisionConstraint {
+	return revisionConstraint{min: &revision}
+}
+
+// validBefore creates a revision constraint that is valid for revisions older than argument.
+func validBefore(revision tosca.Revision) revisionConstraint {
+	return revisionConstraint{max: &revision}
+}
+
+// validInRange creates a revision constraint that is valid only for revision.
+func validOnlyIn(revision tosca.Revision) revisionConstraint {
+	next := revision + 1
+	return revisionConstraint{min: &revision, max: &next}
+}
+
+// validInRange creates a revision constraint that is valid in the range [min, max]
+func validInRange(min, max tosca.Revision) revisionConstraint {
+	next := max + 1
+	return revisionConstraint{min: &min, max: &next}
+}
+
+// generateOpCodesInRange generates a slice of opcodes in the range [start, end].
+func getOpCodesInRange(start, end OpCode) []OpCode {
+	opCodes := make([]OpCode, end-start+1)
+	for i := start; i <= end; i++ {
+		opCodes[i-start] = i
+	}
+	return opCodes
+}
+
+func generateTestCaseFor(f func(op OpCode) testDeclaration, opCodes []OpCode) []testDeclaration {
+	tests := make([]testDeclaration, len(opCodes))
+	for i, op := range opCodes {
+		tests[i] = f(op)
+	}
+	return tests
+}
+
+func toKey(value byte) tosca.Key {
+	res := tosca.Key{}
+	res[len(res)-1] = value
+	return res
+}
+
+func toWord(value int) tosca.Word {
+	v := uint256.NewInt(uint64(value))
+	return v.Bytes32()
+}
+
+type stackTest struct {
+	before    []tosca.Word
+	sizeAfter int
+}
+
+func stackSize(beforeSize, sizeAfter int) stackTest {
+	before := make([]tosca.Word, beforeSize)
+	for i := range before {
+		before[i] = toWord(0x01)
+	}
+	return stackWithValues(before, sizeAfter)
+}
+
+func stackWithValues(before []tosca.Word, sizeAfter int) stackTest {
+	return stackTest{before, sizeAfter}
+}
+
+func invariantStack() stackTest {
+	return stackSize(0, 0)
+}
+
+func cost(gas gas) testParameter[tosca.Gas] {
+	return testParameter[tosca.Gas]{GAS_START, GAS_START - tosca.Gas(gas)}
+}
+
+func attachDataToOp(op OpCode, dataInstructions int) []Instruction {
+	code := []Instruction{{op, 0}}
+	for i := 0; i < dataInstructions; i++ {
+		code = append(code, Instruction{DATA, 0})
+	}
+	return code
+}
+
+func forEachSupportedRevision(t *testing.T, test testDeclaration, f func(t *testing.T, revision tosca.Revision)) {
+	for revision := tosca.R07_Istanbul; revision <= newestSupportedRevision; revision++ {
+		if !test.revisionConstraint.isValidFor(revision) {
+			continue
+		}
+		t.Run(revision.String(), func(t *testing.T) { f(t, revision) })
+	}
+}
+
+func expectOutOfGas() *status {
+	v := statusOutOfGas
+	return &v
+}
+func expectReverted() *status {
+	v := statusReverted
+	return &v
+}
+
+func expectReturned() *status {
+	v := statusReturned
+	return &v
+}
+
+func expectInvalidInstruction() *status {
+	v := statusInvalidInstruction
+	return &v
+}
+
+func makeContext(
```

Comment:
I like this, we have a lot of tests where we insatiate a context and there is always the repeated block.

Reply:
This can be done here because it depends on the testDeclaration type, for other tests this may be quite different.